<?xml version="1.0" encoding="utf-8"?>
<testsuite failures="1" name="SoftwareManagement.sm_apt_install_fail" skipped="0" tests="1" time="43.12" timestamp="2022-03-02T09:05:06">
	<testcase classname="PySysTest" name="SoftwareManagement.sm_apt_install_fail">
		<failure message="BLOCKED: TypeError: 'NoneType' object is not iterable" type="BLOCKED"/>
		<system-out>10:04:22 INFO  ==============================================================
10:04:22 INFO  Id:    SoftwareManagement.sm_apt_install_fail
10:04:22 INFO  Dir:   software_management_end_to_end/
10:04:22 INFO  Title: Validate end to end behaviour for a failing installation
10:04:22 DEBUG Execution order hint: 0.0
10:04:22 INFO  ==============================================================
10:04:22 INFO  Using sw id database: {'asciijump': '5475369', 'robotfindskitten': '5474869', 'squirrel3': '5475279', 'rolldice': '5152439', 'moon-buggy': '5439204', 'apple': '5495053', 'banana': '5494888', 'cherry': '5495382', 'watermelon': '5494510', 'registry': '8018911', 'hello-world': '8021526', 'docker/getting-started': '8021973', 'alpine': '7991792'}
10:04:22 DEBUG EnvironmentC8y Setup
10:04:22 INFO  EnvironmentC8y Setup
10:04:22 DEBUG Process parameters for systemctl&lt;serv_mapper1&gt;
  command line : /usr/bin/systemctl status tedge-mapper-c8y
    arg #1     : status
    arg #2     : tedge-mapper-c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper1.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper1.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
10:04:23 DEBUG Waiting up to 600 secs for process systemctl&lt;serv_mapper1&gt; (pid 12694)
10:04:23 INFO  Executed systemctl&lt;serv_mapper1&gt;, exit status 3
10:04:23 INFO  Restarting mosquitto too frequently in the last 10 seconds. It was only up for 3 seconds
10:04:23 INFO  Delaying execution by 8 seconds
10:04:31 DEBUG Process parameters for sudo&lt;tedge_connect_c8y&gt;
  command line : /usr/bin/sudo /usr/bin/tedge connect c8y
    arg #1     : /usr/bin/tedge
    arg #2     : connect
    arg #3     : c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_connect_c8y.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_connect_c8y.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
10:04:31 DEBUG Waiting up to 600 secs for process sudo&lt;tedge_connect_c8y&gt; (pid 12698)
10:04:41 INFO  Executed sudo&lt;tedge_connect_c8y&gt;, exit status 0
10:04:41 DEBUG Process parameters for sudo&lt;tedge_connect_c8y_test&gt;
  command line : /usr/bin/sudo /usr/bin/tedge connect c8y --test
    arg #1     : /usr/bin/tedge
    arg #2     : connect
    arg #3     : c8y
    arg #4     : --test
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_connect_c8y_test.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_connect_c8y_test.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
10:04:41 DEBUG Waiting up to 600 secs for process sudo&lt;tedge_connect_c8y_test&gt; (pid 12814)
10:04:41 INFO  Executed sudo&lt;tedge_connect_c8y_test&gt;, exit status 0
10:04:41 DEBUG Process parameters for systemctl&lt;serv_mosq2&gt;
  command line : /usr/bin/systemctl status mosquitto
    arg #1     : status
    arg #2     : mosquitto
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mosq2.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mosq2.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
10:04:41 DEBUG Waiting up to 600 secs for process systemctl&lt;serv_mosq2&gt; (pid 12816)
10:04:41 INFO  Executed systemctl&lt;serv_mosq2&gt;, exit status 0
10:04:41 DEBUG Process parameters for systemctl&lt;serv_mapper3&gt;
  command line : /usr/bin/systemctl status tedge-mapper-c8y
    arg #1     : status
    arg #2     : tedge-mapper-c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper3.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper3.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
10:04:41 DEBUG Waiting up to 600 secs for process systemctl&lt;serv_mapper3&gt; (pid 12817)
10:04:41 INFO  Executed systemctl&lt;serv_mapper3&gt;, exit status 0
10:04:43 DEBUG Final URL of the request: https://thin-edge-io.eu-latest.cumulocity.com/devicecontrol/operations?deviceId=14890101&amp;pageSize=1&amp;revert=true&amp;dateFrom=1970-01-01T00%3A00%3A00.000Z
10:04:43 INFO  State of current operation: SUCCESSFUL
10:04:43 DEBUG --- test execute
10:04:44 INFO  Response status: 201
10:04:44 INFO  Response to action: {
    &quot;creationTime&quot;: &quot;2022-03-02T09:04:44.046Z&quot;,
    &quot;deviceId&quot;: &quot;14890101&quot;,
    &quot;deviceName&quot;: &quot;octocatrpi3&quot;,
    &quot;self&quot;: &quot;https://t493319102.eu-latest.cumulocity.com/devicecontrol/operations/14902660&quot;,
    &quot;id&quot;: &quot;14902660&quot;,
    &quot;status&quot;: &quot;PENDING&quot;,
    &quot;description&quot;: &quot;Apply software changes, triggered from PySys: [{'name': 'does_not_exist', 'id': '5446165', 'version': '::apt', 'url': '', 'action': 'install'}]&quot;,
    &quot;c8y_SoftwareUpdate&quot;: [
        {
            &quot;name&quot;: &quot;does_not_exist&quot;,
            &quot;action&quot;: &quot;install&quot;,
            &quot;id&quot;: &quot;5446165&quot;,
            &quot;version&quot;: &quot;::apt&quot;,
            &quot;url&quot;: &quot;&quot;
        }
    ]
}
10:04:44 INFO  Started operation: {'creationTime': '2022-03-02T09:04:44.046Z', 'deviceId': '14890101', 'deviceName': 'octocatrpi3', 'self': 'https://t493319102.eu-latest.cumulocity.com/devicecontrol/operations/14902660', 'id': '14902660', 'status': 'PENDING', 'description': &quot;Apply software changes, triggered from PySys: [{'name': 'does_not_exist', 'id': '5446165', 'version': '::apt', 'url': '', 'action': 'install'}]&quot;, 'c8y_SoftwareUpdate': [{'name': 'does_not_exist', 'action': 'install', 'id': '5446165', 'version': '::apt', 'url': ''}]}
10:04:46 INFO  State of operation 14902660 : EXECUTING
10:04:48 INFO  State of operation 14902660 : EXECUTING
10:04:50 INFO  State of operation 14902660 : EXECUTING
10:04:52 INFO  State of operation 14902660 : EXECUTING
10:04:54 INFO  State of operation 14902660 : EXECUTING
10:04:56 INFO  State of operation 14902660 : EXECUTING
10:04:59 INFO  State of operation 14902660 : EXECUTING
10:05:01 INFO  State of operation 14902660 : EXECUTING
10:05:03 INFO  State of operation 14902660 : FAILED
10:05:03 DEBUG --- test validate
10:05:03 WARN  caught &lt;class 'TypeError'&gt; while running test: 'NoneType' object is not iterable
Traceback (most recent call last):
  File &quot;/home/pi/env-pysys/lib/python3.7/site-packages/pysys/baserunner.py&quot;, line 1211, in __call__
    self.testObj.validate()
  File &quot;/home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/run.py&quot;, line 31, in validate
    &quot;False == value&quot;, value=self.check_is_installed(&quot;does_not_exist&quot;)
  File &quot;/home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/environment_sm_management.py&quot;, line 395, in check_is_installed
    for package in package_list:
TypeError: 'NoneType' object is not iterable
10:05:03 WARN  TypeError: 'NoneType' object is not iterable ... blocked 
10:05:03 DEBUG --- test cleanup
10:05:03 INFO  
10:05:03 INFO  cleanup:
10:05:03 DEBUG Running registered cleanup function: &lt;bound method SoftwareManagement.mysmcleanup of &lt;PySysTest object at 0x75340af0&gt;&gt;
10:05:03 DEBUG Running registered cleanup function: &lt;bound method EnvironmentC8y.myenvcleanup of &lt;PySysTest object at 0x75340af0&gt;&gt;
10:05:03 DEBUG EnvironmentC8y Cleanup
10:05:03 DEBUG Process parameters for sudo&lt;tedge_disconnect_c8y&gt;
  command line : /usr/bin/sudo /usr/bin/tedge disconnect c8y
    arg #1     : /usr/bin/tedge
    arg #2     : disconnect
    arg #3     : c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_disconnect_c8y.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/tedge_disconnect_c8y.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
10:05:03 DEBUG Waiting up to 600 secs for process sudo&lt;tedge_disconnect_c8y&gt; (pid 13056)
10:05:05 INFO  Executed sudo&lt;tedge_disconnect_c8y&gt;, exit status 0
10:05:05 DEBUG Process parameters for systemctl&lt;serv_mapper5&gt;
  command line : /usr/bin/systemctl status tedge-mapper-c8y
    arg #1     : status
    arg #2     : tedge-mapper-c8y
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper5.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/software_management_end_to_end/sm_apt_install_fail/Output/linux/serv_mapper5.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
10:05:05 DEBUG Waiting up to 600 secs for process systemctl&lt;serv_mapper5&gt; (pid 13096)
10:05:06 INFO  Executed systemctl&lt;serv_mapper5&gt;, exit status 3
10:05:06 DEBUG ProcessUser cleanup function done.
10:05:06 INFO  
10:05:06 INFO  Test duration: 43.12 secs
10:05:06 INFO  Test final outcome:  BLOCKED
10:05:06 INFO  Test outcome reason: TypeError: 'NoneType' object is not iterable
10:05:06 INFO  
</system-out>
	</testcase>
</testsuite>
